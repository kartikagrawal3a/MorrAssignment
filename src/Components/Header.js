import React from 'react';
import HeaderCSS from '../styles/Header.module.css';
const Header = (props) => {
    return(
        <>
            <div className={HeaderCSS.header1} title="header1Container">
                <div className="container" title="fixedContainer1">
                    <div className={HeaderCSS.header1Content} title="header1Content">
                        <p title="header1ContentPart1">DEMO Streaming</p>
                        <div className={HeaderCSS.header1ContentPart} title="header1ContentPart">
                            <a href = '/' title="loginLink">Log in</a>
                            <button title="btn">Start your free trail</button>
                        </div>
                    </div>
                </div>
            </div>
            <div className={HeaderCSS.header2} title="header2Container">
                <div className="container" title="fixedContainer2">
                    <div className={HeaderCSS.header2Content} title="header2Content">
                        <p title="header2ContentPart">{`Popular ${props.category}`}</p>
                    </div>
                </div>
            </div>
        </>
    );
}

export default Header;