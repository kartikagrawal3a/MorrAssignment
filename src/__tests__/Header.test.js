import {render, cleanup} from '@testing-library/react';
import renderer from 'react-test-renderer';
import Header from '../Components/Header';

afterEach(()=>{
    cleanup();
})

test("checkHeaderRender1", ()=>{
    const cat = "Titles";
    const { getByTitle} = render(<Header category = {cat} />);
    getByTitle("header1Container");
    getByTitle("fixedContainer1");
    getByTitle("header1Content");
    const para = getByTitle("header1ContentPart1");
    expect(para).toHaveTextContent("DEMO Streaming")
    getByTitle("header1ContentPart");
    const loginLink = getByTitle("loginLink");
    expect(loginLink).toHaveTextContent("Log in");
    const btn  = getByTitle("btn");
    expect(btn).toHaveTextContent("Start your free trail");
    getByTitle("header2Container");
    getByTitle("fixedContainer2");
    getByTitle("header2Content");
    const para2 = getByTitle("header2ContentPart");
    expect(para2).toHaveTextContent("Popular Titles");
})

test("checkHeaderRender2", ()=>{
    const cat = "Movies";
    const { getByTitle} = render(<Header category = {cat} />);
    getByTitle("header1Container");
    getByTitle("fixedContainer1");
    getByTitle("header1Content");
    const para = getByTitle("header1ContentPart1");
    expect(para).toHaveTextContent("DEMO Streaming");
    getByTitle("header1ContentPart");
    const loginLink = getByTitle("loginLink");
    expect(loginLink).toHaveTextContent("Log in");
    const btn  = getByTitle("btn");
    expect(btn).toHaveTextContent("Start your free trail");
    getByTitle("header2Container");
    getByTitle("fixedContainer2");
    getByTitle("header2Content");
    const para2 = getByTitle("header2ContentPart");
    expect(para2).toHaveTextContent("Popular Movies");
})

test("checkHeaderRender3", ()=>{
    const cat = "Series";
    const { getByTitle} = render(<Header category = {cat} />);
    getByTitle("header1Container");
    getByTitle("fixedContainer1");
    getByTitle("header1Content");
    const para = getByTitle("header1ContentPart1");
    expect(para).toHaveTextContent("DEMO Streaming");
    getByTitle("header1ContentPart");
    const loginLink = getByTitle("loginLink");
    expect(loginLink).toHaveTextContent("Log in");
    const btn  = getByTitle("btn");
    expect(btn).toHaveTextContent("Start your free trail");
    getByTitle("header2Container");
    getByTitle("fixedContainer2");
    getByTitle("header2Content");
    const para2 = getByTitle("header2ContentPart");
    expect(para2).toHaveTextContent("Popular Series");
})

test('match snapshot #1', ()=>{
    const cat = "Titles";
    const tree = renderer.create(<Header category = {cat} />).toJSON();
    expect(tree).toMatchSnapshot();
})

test('match snapshot #2', ()=>{
    const cat = "Series";
    const tree = renderer.create(<Header category = {cat} />).toJSON();
    expect(tree).toMatchSnapshot();
})

test('match snapshot #3', ()=>{
    const cat = "Movies";
    const tree = renderer.create(<Header category = {cat} />).toJSON();
    expect(tree).toMatchSnapshot();
})