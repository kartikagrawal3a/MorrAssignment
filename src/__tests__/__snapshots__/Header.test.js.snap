// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`match snapshot #1 1`] = `
Array [
  <div
    className="header1"
    title="header1Container"
  >
    <div
      className="container"
      title="fixedContainer1"
    >
      <div
        className="header1Content"
        title="header1Content"
      >
        <p
          title="header1ContentPart1"
        >
          DEMO Streaming
        </p>
        <div
          className="header1ContentPart"
          title="header1ContentPart"
        >
          <a
            href="/"
            title="loginLink"
          >
            Log in
          </a>
          <button
            title="btn"
          >
            Start your free trail
          </button>
        </div>
      </div>
    </div>
  </div>,
  <div
    className="header2"
    title="header2Container"
  >
    <div
      className="container"
      title="fixedContainer2"
    >
      <div
        className="header2Content"
        title="header2Content"
      >
        <p
          title="header2ContentPart"
        >
          Popular Titles
        </p>
      </div>
    </div>
  </div>,
]
`;

exports[`match snapshot #2 1`] = `
Array [
  <div
    className="header1"
    title="header1Container"
  >
    <div
      className="container"
      title="fixedContainer1"
    >
      <div
        className="header1Content"
        title="header1Content"
      >
        <p
          title="header1ContentPart1"
        >
          DEMO Streaming
        </p>
        <div
          className="header1ContentPart"
          title="header1ContentPart"
        >
          <a
            href="/"
            title="loginLink"
          >
            Log in
          </a>
          <button
            title="btn"
          >
            Start your free trail
          </button>
        </div>
      </div>
    </div>
  </div>,
  <div
    className="header2"
    title="header2Container"
  >
    <div
      className="container"
      title="fixedContainer2"
    >
      <div
        className="header2Content"
        title="header2Content"
      >
        <p
          title="header2ContentPart"
        >
          Popular Series
        </p>
      </div>
    </div>
  </div>,
]
`;

exports[`match snapshot #3 1`] = `
Array [
  <div
    className="header1"
    title="header1Container"
  >
    <div
      className="container"
      title="fixedContainer1"
    >
      <div
        className="header1Content"
        title="header1Content"
      >
        <p
          title="header1ContentPart1"
        >
          DEMO Streaming
        </p>
        <div
          className="header1ContentPart"
          title="header1ContentPart"
        >
          <a
            href="/"
            title="loginLink"
          >
            Log in
          </a>
          <button
            title="btn"
          >
            Start your free trail
          </button>
        </div>
      </div>
    </div>
  </div>,
  <div
    className="header2"
    title="header2Container"
  >
    <div
      className="container"
      title="fixedContainer2"
    >
      <div
        className="header2Content"
        title="header2Content"
      >
        <p
          title="header2ContentPart"
        >
          Popular Movies
        </p>
      </div>
    </div>
  </div>,
]
`;
